config/confighttp/confighttp_test.go

```diff
@@ -381,10 +381,14 @@ func TestHttpReception(t *testing.T) {
 			}
 			ln, err := hss.ToListener()
 			assert.NoError(t, err)
-			s := hss.ToServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				_, errWrite := fmt.Fprint(w, "test")
-				assert.NoError(t, errWrite)
-			}), componenttest.NewNopTelemetrySettings())
+			s, err := hss.ToServer(
+				componenttest.NewNopHost(),
+				componenttest.NewNopTelemetrySettings(),
+				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					_, errWrite := fmt.Fprint(w, "test")
+					assert.NoError(t, errWrite)
+				}))
+			assert.NoError(t, err)
```

Comment:
require seems more appropriate since we use `s` below
```suggestion
			require.NoError(t, err)
```

Reply:
Sounds reasonable. I fixed the `assert.NoError` present earlier in the same context to use `require.NoError`.